ivy/functional/backends/torch/general.py

```diff
@@ -190,28 +190,78 @@ def gather(
     batch_dims %= len(params.shape)
     ivy.utils.assertions.check_gather_input_valid(params, indices, axis, batch_dims)
     result = []
+
+    def expand_p_i(params, indices, axis=axis, batch_dims=batch_dims):
+        axis %= len(params.shape)
+        indices_ex = torch.clone(indices).detach()
```

Comment:
not sure why we're detaching the indices here, probably it's an integer tensor anyway right? Sorry to be a pain, but given that this is a very important function, we'd need to be quite careful before merging any changes to it 😅

Reply:
Not a problem at all- in this case, it's probably unnecessary, but as quite often with tensors/arrays, it's easy to accidentally alter the input which is meant to be unchanged, so when in doubt, I write the code with cloning first. In this case, the cloning can probably be dropped whatsoever and `indices_ex` can probably get replaced with just `indices` within that function. I'll do it and check if nothing breaks. Thanks for pointing that out :)

Reply:
@vedpatwardhan Changes applied, tests passed for `num_examples=200` but I am not discouraging you from running pytest separately :)